src/index.ts

```diff
@@ -39,22 +39,69 @@ function logError(msg) {
   spinner.fail();
 }
 
-function transformWebModuleFilename(depName: string): string {
-  return depName.replace('/', '--');
+class ErrorWithHint extends Error {
+  constructor(message: string, public readonly hint: string) {
```

Comment:
You've not restored the prototype chain. does `err instanceof ErrorWithHint` work correctly?

(this is what I did in the original pull request, but I don't know whether it's still necessary in modern versions of TypeScript):
https://github.com/pikapkg/web/pull/21/files#diff-f41e9d04a45c83f3b6f6e630f10117feR71

Reply:
Yea I couldn't find a modern TS answer either (the link in the comment was a 404 when I tried it). 

I'm comfortable just doing a check for the `hint` property on the error object, since TS doesn't even allow you to type exceptions. But if you would prefer that we do an instanceof check it would be good to have a definitive answer to make sure that workaround is necessary.

Reply:
> just doing a check for the `hint` property
yup, that's acceptable. duck-typing is close enough.